pkg/trace/sampler/config_watcher.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2015 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package sampler
```

Comment:
The content of this file is completely untested, I don't know if we have any recommended way of testing that we correctly set metrics or if it's ok to leave things like that

Reply:
I took a glance and it seems like it should be possible to have your own Mock version of the metrics client that you step on in the `metrics` package and can check to ensure it's called as expected. I would definitely add a couple tests here to ensure you get what you expect (Although at first glance this all looks pretty good)